From 2f2aaaa81330ff540f48317f4c9eed0a427a6445 Mon Sep 17 00:00:00 2001
From: maverick <maverick_jia@sina.com>
Date: Sun, 30 Jul 2023 11:00:31 +0800
Subject: [PATCH 16/45] rotate screen for nabu

---
 drivers/video/fbdev/core/fbcon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/core/fbcon.c b/drivers/video/fbdev/core/fbcon.c
index 14498a0d1..07fc4881b 100644
--- a/drivers/video/fbdev/core/fbcon.c
+++ b/drivers/video/fbdev/core/fbcon.c
@@ -157,7 +157,7 @@ static char fontname[40];
 static int info_idx = -1;
 
 /* console rotation */
-static int initial_rotation = -1;
+static int initial_rotation = 1;
 static int fbcon_has_sysfs;
 static int margin_color;
 
-- 
2.34.1

