From c515aa3b4823363ed3da2f4cac4b214a874b0be9 Mon Sep 17 00:00:00 2001
From: maverick <maverick_jia@sina.com>
Date: Sun, 12 Nov 2023 06:44:24 +0800
Subject: [PATCH 22/44] NABU: add backlight regulators

---
 .../sm8150-xiaomi-nabu-maverick-minimal.dts   | 26 +++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sm8150-xiaomi-nabu-maverick-minimal.dts b/arch/arm64/boot/dts/qcom/sm8150-xiaomi-nabu-maverick-minimal.dts
index 99f307c87..de34439c8 100644
--- a/arch/arm64/boot/dts/qcom/sm8150-xiaomi-nabu-maverick-minimal.dts
+++ b/arch/arm64/boot/dts/qcom/sm8150-xiaomi-nabu-maverick-minimal.dts
@@ -165,6 +165,32 @@ vreg_s4a_1p8: pm8150-s4 {
 
 		vin-supply = <&vph_pwr>;
 	};
+
+	lcd_avdd_5p5: lcd-avdd-regulator {
+		compatible = "regulator-fixed";
+		regulator-name = "lcd_avdd_5p5";
+
+		regulator-min-microvolt = <5500000>;
+		regulator-max-microvolt = <5500000>;
+		regulator-enable-ramp-delay = <233>;
+		gpio = <&tlmm 114 0>;
+		enable-active-high;
+		regulator-always-on;
+		regulator-boot-on;
+	};
+
+	lcd_avee_5p5: lcd-avee-regulator {
+		compatible = "regulator-fixed";
+		regulator-name = "lcd_avee_5p5";
+
+		regulator-min-microvolt = <5500000>;
+		regulator-max-microvolt = <5500000>;
+		regulator-enable-ramp-delay = <233>;
+		gpio = <&tlmm 115 0>;
+		enable-active-high;
+		regulator-always-on;
+		regulator-boot-on;
+	};
 };
 
 &pm8150_gpios {
-- 
2.47.0

